Frameworks/CoreGraphics/CGColor.mm

```diff
@@ -62,18 +68,20 @@ CGColorRef CGColorRetain(CGColorRef color) {
 }
 
 /**
- @Status Interoperable
+ @Status Caveat
+ @Notes only supports RGB
 */
 CGColorRef CGColorCreate(CGColorSpaceRef colorSpace, const CGFloat* components) {
-    CGColorRef ret =
-        (CGColorRef)[[__LazyUIColor colorWithRed:components[0] green:components[1] blue:components[2] alpha:components[3]] retain];
-
+    RETURN_NULL_IF(!colorSpace);
+    RETURN_NULL_IF(!components);
+    CGColorRef ret = static_cast<CGColorRef>(
```

Comment:
it might be safe to assert that the #components in the color space is 3-4. Just so that we don't index beyond the end of components.